llvm/include/llvm/ADT/APInt.h

```diff
@@ -1883,6 +1875,12 @@ class [[nodiscard]] APInt {
   /// Returns whether this instance allocated memory.
   bool needsCleanup() const { return !isSingleWord(); }
 
+  /// Get the word corresponding to a bit position
+  /// \returns the corresponding word for the specified bit position.
+  uint64_t getWord(unsigned bitPosition) const {
```

Comment:
Would it be possible to avoid adding this by using `LowHalf == APInt.getWord(0) == APint.trunc(64).getZExtValue()` and `HighHalf == APInt.getWord(64) == APInt.lshr(64).trunc(64).getZExtValue()`

Reply:
This is what the extractBitsAsZExtValue API is for.

Reply:
extractBitsAsZExtValue is much better, thanks.